r/tests/testthat/test-RecordBatch.R

```diff
@@ -642,14 +642,19 @@ test_that("Handling string data with embedded nuls", {
   batch_with_nul$b <- batch_with_nul$b$cast(utf8())
 
   withr::with_options(list(arrow.skip_nul = TRUE), {
-    expect_warning(
-      expect_equal(
-        as.data.frame(batch_with_nul)$b,
-        c("person", "woman", "man", "camera", "tv"),
-        ignore_attr = TRUE
-      ),
-      "Stripping '\\0' (nul) from character vector",
-      fixed = TRUE
+    # Because expect_equal() may call identical(x, y) more than once,
```

Comment:
"may"? Is it not deterministic? Why would it do the work multiple times?

Would it help to change the test, like `expect_true(all(as.data.frame(batch_with_nul)$b == c("person", "woman", "man", "camera", "tv")))`?